### PR TITLE
Update gradle wrapper and fix gradle deprecation warnings.

### PR DESCRIPTION
**Pull request by jameshilliard**
This is a WIP attempt at fixing gradle 6 compatibility.
When trying to build this with a project that uses gradle 6 I get the following build error:
```
An exception occurred applying plugin request [id: 'com.devsoap.plugin.vaadin', version: '2.0.0.beta1']
> Failed to apply plugin [id 'com.devsoap.plugin.vaadin']
   > Could not create an instance of type com.devsoap.plugin.extensions.TestBenchNodeExtension.
      > Please use the ObjectFactory.listProperty() method to create a property of type List<T>.
```

```
Caused by: org.gradle.api.InvalidUserCodeException: Please use the ObjectFactory.listProperty() method to create a property of type List<T>.
	at org.gradle.api.internal.model.DefaultObjectFactory.property(DefaultObjectFactory.java:135)
	at org.gradle.api.model.ObjectFactory$property.call(Unknown Source)
	at com.devsoap.plugin.extensions.TestBenchNodeExtension.<init>(TestBenchNodeExtension.groovy:44)
	at com.devsoap.plugin.extensions.TestBenchNodeExtension_Decorated.<init>(Unknown Source)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at org.gradle.internal.instantiation.generator.AsmBackedClassGenerator.newInstance(AsmBackedClassGenerator.java:200)
	at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.newInstance(AbstractClassGenerator.java:402)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:64)
```

### CODE REVIEW

**Comment by johndevs** [comment]
Since the wrapper is updated to 5.6 this will mean that the minimum version of Gradle will be bumped to that version as the plugin will be compiled for that version. 

People with Gradle 5.5 and below will no longer be able to use the plugin which might be an issue as they would be getting runtime issues when calling the new methods.

To reflect this change I would also bump the minimum gradle version in https://github.com/johndevs/gradle-vaadin-plugin/blob/4376ea80eec4ad7da88dfedc2ecaf855e79168d3/src/main/groovy/com/devsoap/plugin/GradleVaadinPlugin.groovy#L215 to 5.6 so it doesn't come as a surprise to users.

File: src/main/groovy/com/devsoap/plugin/GradleVaadinPlugin.groovy [hunk 1]
```diff
@@ -212,7 +212,7 @@ class GradleVaadinPlugin implements Plugin<Project> {
 
         Gradle gradle = project.gradle
         VersionNumber version = VersionNumber.parse(gradle.gradleVersion)
-        VersionNumber requiredVersion = new VersionNumber(5, 0, 0, null)
+        VersionNumber requiredVersion = new VersionNumber(5, 6, 2, null)
```
Comment (jameshilliard) [inline src/main/groovy/com/devsoap/plugin/GradleVaadinPlugin.groovy]: updated

**Comment by jameshilliard** [comment]
@johndevs I finally got all tests passing in travis looks like, anything else needed or should this be good to merge now?

**Comment by johndevs** [comment]
@jameshilliard Yes, this looks good. I'll still do some manual tests with this for some of the corner-cases the tests won't cover and then if all looks good I'll merge it. 

Thanks for the hard work :+1:

**Comment by jameshilliard** [comment]
@johndevs think you could also upload a new maven release for this?

**Comment by johndevs** [comment]
@jameshilliard Yes, I'll release 2.0.0.beta2 this weekend with your changes.
